Give schedules access to the key, step, and last eval score (#804)
explosion/thinc: website/docs/usage-training.md

A common trick for stochastic gradient descent is to **vary the learning rate or
other hyperparameters** over the course of training. Since there are many
possible ways to vary the learning rate, Thinc lets you implement hyperparameter
schedules as simple callback functions. Thinc also provides a number of
[popular schedules](/docs/api-schedules) built-in.

You can use schedules directly, by calling the schedule with the `step` keyword
argument and using it to update hyperparameters in your training loop. Since
schedules are particularly common for optimization settings, the
[`Optimizer`](/docs/api-optimizer) object accepts floats, lists, iterators, and
schedule callbacks for most of its parameters. When you call
[`Optimizer.step_schedules`](/docs/api-optimizer#step_schedules), the optimizer

> will increase its step count and pass this step count to the schedules. For

Reviewer: ```suggestion
will increase its step count and pass it to the schedules. For
```

Author: Fixed.

> instance, here's how to create an instance of the `Adam` optimizer with a custom

Reviewer: ```suggestion
instance, this is how one creates an instance of the `Adam` optimizer with a custom
```
This sounds slightly better, I find.

Author: Fixed.
